backend/internal/usecase/transaction.go

```diff
@@ -145,3 +167,117 @@ func (uc *TransactionUsecase) QueryIncludedTransactions(ctx context.Context, txL
 	}
 	return txs, nil
 }
+
+func (uc *TransactionUsecase) QueryTransactionDetail(ctx context.Context, txHash string) (*QueryTransactionDetailResp, *error.Http) {
+	txHashBytes, err := hex.DecodeString(txHash)
+	if err != nil {
+		err := error.NewHttpError(
+			"unable to decode tx hash",
+			"valid txHash query parameter is required",
+			http.StatusBadRequest,
+		)
+		return nil, &err
+	}
+
+	erpcTX, erpcTxErr := uc.erpcDBQuery.QueryFromTransactionDetail(ctx, txHash)
+	if erpcTxErr != nil && erpcTxErr != pgx.ErrNoRows {
+		log.Err(err).Msg("err encountered while querying erpc DB")
+		err := error.NewHttpError(
+			"error encountered while querying for data",
+			"",
+			http.StatusInternalServerError,
+		)
+		return nil, &err
+	}
+
+	tse, err := uc.observerDBQuery.QueryDecryptedTXFromSubmittedEvent(ctx, txHashBytes)
```

Comment:
IK think this query will miss on a certain case. 

If the user requested the user transaction and the key hasn't been released yet, it will not find the txhash, even though the ERPC provided us with the sequencer tx hash for this user tx hash.